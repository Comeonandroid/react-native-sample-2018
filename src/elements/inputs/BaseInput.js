import React, { PureComponent } from 'react'
import { MKTextField } from 'react-native-material-kit'
import { PropTypes } from 'prop-types'
import { withTheme } from 'styled-components'

class BaseInput extends PureComponent {

  constructor (props) {
    super(props)

    this.handleTextChange = this.handleTextChange.bind(this)
    this.handleBlur = this.handleBlur.bind(this)
  }

  handleBlur() {
    const { name, onBlur } = this.props
    if (onBlur) {
      onBlur(name, true)
    }
  }

  handleTextChange(value) {
    this.props.onChange(this.props.name, value)
  }

  render() {
    const {
      placeholder,
      textInputStyle,
      tintColor,
      highlightColor,
      underlineSize,
      selectionColor,
      theme,
      ...rest } = this.props // eslint-disable-line no-unused-vars
    return (
      <MKTextField
        {...rest}
        tintColor={tintColor}
        highlightColor={highlightColor}
        underlineSize={underlineSize}
        textInputStyle={textInputStyle}
        placeholder={placeholder}
        onTextChange={this.handleTextChange}
        onBlur={this.handleBlur}
        selectionColor={selectionColor || theme.colors.primary}
      />
    )
  }
}

BaseInput.propTypes = {
  name: PropTypes.string,
  placeholder: PropTypes.string,
  textInputStyle: PropTypes.object,
  tintColor: PropTypes.string,
  keyboardType: PropTypes.string,
  highlightColor: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  underlineSize: PropTypes.number,
  selectionColor: PropTypes.string,
  theme: PropTypes.object
}

BaseInput.defaultProps = {
  underlineSize: 1
}

export default withTheme(BaseInput)
