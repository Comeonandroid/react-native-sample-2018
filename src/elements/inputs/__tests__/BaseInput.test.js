import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../specs/shared_examples/index'

import BaseInput from '../BaseInput'

const setup = propOverrides => {
  const props = Object.assign({
    name: 'name',
    value: 'value',
    children: null,
    onChange: jest.fn()
  }, propOverrides)

  const component = <BaseInput {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<BaseInput />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
