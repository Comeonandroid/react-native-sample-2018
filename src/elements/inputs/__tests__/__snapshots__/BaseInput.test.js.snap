// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> dark: renders correctly 1`] = `
<View
  style={
    Object {
      "height": 39,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onChangeText={[Function]}
    onFocus={[Function]}
    secureTextEntry={undefined}
    selectionColor="#faa632"
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "transparent",
          "marginTop": 0,
          "paddingBottom": 2,
          "paddingTop": 2,
        },
        Object {
          "color": "rgb(117, 117, 117)",
          "fontSize": 16,
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value="value"
  />
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 1,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "rgba(63,81,181, 0.9)",
          "height": 1,
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<BaseInput /> light: renders correctly 1`] = `
<View
  style={
    Object {
      "height": 39,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onChangeText={[Function]}
    onFocus={[Function]}
    secureTextEntry={undefined}
    selectionColor="#faa632"
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "transparent",
          "marginTop": 0,
          "paddingBottom": 2,
          "paddingTop": 2,
        },
        Object {
          "color": "rgb(117, 117, 117)",
          "fontSize": 16,
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value="value"
  />
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 1,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "rgba(63,81,181, 0.9)",
          "height": 1,
          "left": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
  </View>
</View>
`;
